DeepCat.js

```diff
@@ -11,7 +11,20 @@
 	var DBname = mw.config.get( 'wgDBname' );
 	var requestUrl = '//tools.wmflabs.org/catgraph-jsonp/' + DBname + '_ns14/traverse-successors%20Category:{0}%20' + maxDepth + '%20' + maxResults;
 
-
+	// hash function for generating hint box cookie token.
+	// see http://erlycoder.com/49/javascript-hash-functions-to-convert-string-into-integer-hash-
+	function djb2Code(str) {
```

Comment:
I do not know anything better in JavaScript, no. DJB2 (that's actually the name of the hash function, I did not know that before) looks fine.